src/Umbraco.Web/PublishedContentExtensions.cs

```diff
@@ -20,115 +21,46 @@ namespace Umbraco.Web
+        public static bool IsComposedOf(this IPublishedContent content, string alias)
         {
-            return content.Url;
+            return content.ContentType.CompositionAliases.Contains(alias);
         }
 
-        /// <summary>
-        /// Gets the absolute url for the content.
-        /// </summary>
-        /// <param name="content">The content.</param>
-        /// <returns>The absolute url for the content.</returns>
-        public static string UrlAbsolute(this IPublishedContent content) => content.UrlAbsolute(null);
+        #endregion
+
+        #region Template
 
         /// <summary>
-        /// Gets the absolute url for the content.
+        /// Returns the current template Alias
         /// </summary>
-        /// <param name="content">The content.</param>
-        /// <param name="culture">The culture to get the url for</param>
-        /// <returns>The absolute url for the content.</returns>
-        public static string UrlAbsolute(this IPublishedContent content, string culture)
+        /// <param name="content"></param>
+        /// <returns>Empty string if none is set.</returns>
+        public static string GetTemplateAlias(this IPublishedContent content)
         {
-            // adapted from PublishedContentBase.Url
-
-            if (Current.UmbracoContext == null)
-                throw new InvalidOperationException("Cannot resolve a Url for a content item when Current.UmbracoContext is null.");
-            if (Current.UmbracoContext.UrlProvider == null)
-                throw new InvalidOperationException("Cannot resolve a Url for a content item when Current.UmbracoContext.UrlProvider is null.");
-
-            switch (content.ItemType)
+            if(content.TemplateId.HasValue == false)
             {
-                case PublishedItemType.Content:
-                    return Current.UmbracoContext.UrlProvider.GetUrl(content.Id, true, culture);
-                case PublishedItemType.Media:
-                    return Current.UmbracoContext.UrlProvider.GetMediaUrl(content, Constants.Conventions.Media.File, true);
-                default:
-                    throw new ArgumentOutOfRangeException();
+                return string.Empty;
             }
-        }
-
-        /// <summary>
-        /// Gets the url for the media.
-        /// </summary>
-        /// <param name="content">The content.</param>
-        /// <param name="propertyAlias">The property alias to resolve the url from.</param>
-        /// <param name="culture">The variation language.</param>
-        /// <returns>The url for the content.</returns>
-        /// <remarks>Better use the <c>GetMediaUrl</c> method but that method is here to complement <c>MediaUrlAbsolute()</c>.</remarks>
-        public static string MediaUrl(this IPublishedContent content, string propertyAlias, string culture = null)
```

Comment:
Took me a while to figure this one out but this has caused a bug during the 8.1 development. These methods were added in this PR: https://github.com/umbraco/Umbraco-CMS/pull/5282/files and in that PR some things we're using the MediaUrl ext method, namely `ImageCropperTemplateExtensions.GetCropUrl`, but with these changes, this wasn't taken into account and the new `MediaUrl` ext method in this PR has the parameters reversed. I'll look further in this PR for anything that was calling into the old MediaUrl ext method since the params will need to be updated.